src/main/java/bio/terra/cli/app/ToolsManager.java

```diff
@@ -0,0 +1,304 @@
+package bio.terra.cli.app;
+
+import bio.terra.cli.model.GlobalContext;
+import bio.terra.cli.model.TerraUser;
+import bio.terra.cli.model.WorkspaceContext;
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.async.ResultCallback;
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.command.CreateContainerResponse;
+import com.github.dockerjava.api.command.WaitContainerResultCallback;
+import com.github.dockerjava.api.exception.NotFoundException;
+import com.github.dockerjava.api.model.Bind;
+import com.github.dockerjava.api.model.Frame;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.Volume;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.httpclient5.ApacheDockerHttpClient;
+import com.github.dockerjava.transport.DockerHttpClient;
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class runs client-side tools and manipulates the tools-related properties of the global
+ * context object.
+ */
+public class ToolsManager {
+  private static final Logger logger = LoggerFactory.getLogger(ToolsManager.class);
+
+  private final GlobalContext globalContext;
+  private final WorkspaceContext workspaceContext;
+  private DockerClient dockerClient;
+
+  // This is where the pet key files will be mounted on the Docker container.
+  public static final String PET_KEYS_MOUNT_POINT = "/usr/local/etc/terra_cli";
+
+  public ToolsManager(GlobalContext globalContext, WorkspaceContext workspaceContext) {
+    this.globalContext = globalContext;
+    this.workspaceContext = workspaceContext;
+    this.dockerClient = null;
+  }
+
+  // This variable specifies the default Docker image id or tag that the CLI uses to run external
+  // tools.
+  // TODO: change this to a DockerHub or GCR path
+  private static final String DEFAULT_DOCKER_IMAGE_ID = "terra/cli:v0.0";
+
+  /** Returns the default image id. */
+  public static String defaultImageId() {
+    return DEFAULT_DOCKER_IMAGE_ID;
+  }
+
+  /**
+   * Update the Docker image property of the global context.
+   *
+   * @return true if the Docker image property was updated, false otherwise
+   */
+  public boolean updateImageId(String imageId) {
```

Comment:
This is useful for debugging / dev iteration, so you don't have to change a hard-coded string and re-compile every time you modify the Docker image. I also thought about backwards compatibility as a possible use case here, but not sure if that would be better done at a higher level (i.e. Terra CLI version 1.2 means docker image tag ABC, not setting them separately).

Similar to the app execute command, I added a [FOR DEBUG] qualifier to the description for now so we know to revisit this with product. My preference is to keep this in for now, at least while we're iterating with the solutions team. Once I get installation/distribution working, this might be helpful for them also to test things out.